### PR TITLE
[INDEXING-78] Prevent heights from being skipped

### PR DESCRIPTION
**Pull request by deelawn**
I've seen it happen a few times now where some kind of adverse network event can cause last data processing to halt for some period of time.  When it comes back online, the difference between the last height recorded and the most recent height from the network exceeds the maximum heights to process value, effectively causing everything in between to be missed.  It might be better to let it catch up on its own rather than have to monitor for skipped ranges.

### CODE REVIEW

File: manager/client/last.go [hunk 1]
```diff
@@ -896,8 +896,8 @@ func getLastHeightRange(lastKnownHeight, maximumHeightsToGet, lastBlockFromNetwo
 
 	if maximumHeightsToGet < lastBlockFromNetwork-lastKnownHeight {
 		return shared.HeightRange{
-			StartHeight: lastBlockFromNetwork - maximumHeightsToGet,
-			EndHeight:   lastBlockFromNetwork,
+			StartHeight: lastKnownHeight,
```
Comment (pacmessica) [inline manager/client/last.go:899]: should this be lastKnownHeight+1?

**Review by pacmessica — COMMENTED** [review]
oh man definitely had this issue too. For cases where syncing stops for a short period of time, this is definitely handy.

 A few considerations though:
1) What happens when `lastKnownHeight` is 0? If starting a sync for the first time, I think `lastdata` runner should start syncing latest data and not from the beginning of the chain.  

2) if there's a problematic height, then this means `latest` will be stuck indefinitely retrying the exact same height range. 

3) If syncing gets stuck like above, it's not straightforward to skip the problematic height and resume syncing. At least the current implementation can 'unstick' itself and continue collecting latest data, and then we have time to debug the problematic height 

so I'm not sure this should be the default behaviour, maybe it makes sense as a config option though?

**Comment by lukanus** [comment]
After long I had enough time to click, and seen it's only one line, lol. Sorry. 
I'd say this - initially I've set this 2 processes to run in parallel. 
So that initial process may be running and scraping the data, and second one just fetch the latest, within the frame. Although...  we may as well treat it as always the latest from the scheduler without any frame. 
This would obviously stop on a network failure, but at least nothing would be missed on failure

**Comment by deelawn** [comment]
Will update go.mod with new indexing-engine tag after that PR is merged.

**Review by lukanus — APPROVED** [review]

**Pull request merged**